components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,836 @@
+            return dataArray.every(element => {
+                return parseInt(element) >= val && parseInt(element) <= endVal;
+            });
+        };
+
+        var validateDayOfMonthValue = function(dayOfMonth, dayOfWeek) {
+            if(dayOfWeek === '?' && dayOfMonth.includes('L')) {
+                if(dayOfMonth.includes(',') && dayOfWeek === '?'){
+                    return false;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+
+                    return dayOfMonthRangeArr[0] === 'L' && validateRangeOfDay([dayOfMonthRangeArr[1]], 1, 30);
+                }
+            } else if(dayOfMonth.includes('W') && dayOfWeek === '?'){
+                if(dayOfMonth.search(constants.REGEX_FOR_ALPHA_NUMERIC) !== -1 ){
+                    errorMsg = "Invalid Cron Expression";
+                    return false;
+                }
+                if(dayOfMonth.length === 1){
+                    errorMsg = "Invalid Cron Expression";
+                    return false;
+                } else {
+                    errorMsg = "Day of month values must be between 1 and 31";
+                    return validateRangeOfDay([dayOfMonth], 1, 31);
+                }
+            }
+            if (dayOfMonth.search(constants.REGEX_FOR_DAYOFMONTH) !== -1) {
+                errorMsg = "Invalid Cron Expression";
+                return false;
+            }
+            var list = dayOfMonth.split(",");
+            return list.every(function (dayOfMonth){
+                if(dayOfMonth.includes('/') && dayOfWeek === '?') {
+                    if(dayOfMonth.endsWith('/')){
+                        errorMsg = "Day of month values should not end with /";
+                    } else {
+                        errorMsg = "Day of month values must be between 1 and 31";
+                    }
+                    var startingDayOfMonthOptionArr = dayOfMonth.split('/');
+                    var isValidElements = (validateRangeOfDay([startingDayOfMonthOptionArr[0]], 1, 31) &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 31));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 31));
+                    return isValidElements || isValidFirstElem;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+                    if(dayOfMonth.endsWith('-')){
+                        errorMsg = "Day of month values should not end with - ";
+                    } else {
+                        errorMsg = "Day of month values must be between 1 and 31";
+                    }
+                    return validateRangeOfDay(dayOfMonthRangeArr, 1, 31) &&
```

Comment:
Why you need to pass hard coded values like 1, 31 for validateRangeOfDay. Can this 1, 31 can be different in some other case ??

Reply:
In here, I used to check the month values

Reply:
ok